### PR TITLE
Change jsonrpc default protocol to "2.0"

### PR DESCRIPTION
**Pull request by cboulanger**
This is BC because anyone who still uses the original "qx1" protocol can switch back by setting the "protocol" property accordingly.

### CODE REVIEW

**Review by johnspackman — APPROVED** [review]

**Review by level420 — APPROVED** [review]

File: framework/source/class/qx/io/remote/Rpc.js [hunk 1]
```diff
@@ -390,7 +390,7 @@ qx.Class.define("qx.io.remote.Rpc",
      */
     protocol :
     {
-      init : "qx1",
+      init : "2.0",
```
Comment (level420) [inline framework/source/class/qx/io/remote/Rpc.js]: I'm not using it, but this is a BC breaking change. Or did I misunderstand something?

Comment (cboulanger) [reply framework/source/class/qx/io/remote/Rpc.js]: It's a breaking change but can be easily made BC by setting the property. I don't think many people are using the legacy mode. We can put it into the relase notes under "breaking changes".

Comment (oetiker) [reply framework/source/class/qx/io/remote/Rpc.js]: has there been any other mode than the qx1 mode ?

Comment (oetiker) [reply framework/source/class/qx/io/remote/Rpc.js]: all our apps use qx jsonrpc

**Review by hkollmann — COMMENTED** [review]
Our's too. In qx1 mode.

**Comment by cboulanger** [comment]
Actually, I don't care so much - we can leave it in "qx1" mode, given that I will add a wholly separate jsonrpc v2 client. I thought the idea was to deprecate "qx1" mode since it is non standard-compliant...

**Comment by hkollmann** [comment]
Is this change really necessary for your new implementation? Otherwise i would prefer not to change the default

**Comment by cboulanger** [comment]
@hkollmann No it is not necessary. I had the (wrong) impression that we agreed to deprecate "qx1". I can close this, no problem!

**Comment by hkollmann** [comment]
Thanks, We can deprecate it with your new implementation. IMHO a lot of older programs use qx1 - they are programmed before rpc2 exists.

**Comment by derrell** [comment]
We're not releasing the new code as part of the upcoming 6.0 release. We don't want new code using the qx1 variant of JSON RPC; we want them using 2.0. I feel pretty strongly that we should, in fact, make this change, and document it in the release notes. It's a trivial change to existing code, and gives people a JSON RPC 2.0 client they can use as 6.0 comes out the door.

**Review by hkollmann — APPROVED** [review]

**Comment by cboulanger** [comment]
I probably need to update the tests for "qx1" mode.

**Review by derrell — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Comment by oetiker** [comment]
so with this change we are activating a protocol implementation that is not used and is going to be replaced with a new implementation anyway ? right ?

**Comment by derrell** [comment]
@oetiker I'm not quite sure where you got that, but no, not right. JSON-RPC 2.0 is many years old and has been in extensive use. There are lots and lots of existing JSON-RPC 2.0 servers out there, and the 2.0 client code has been in use in qooxdoo for a number of years. I expect that the majority of users of qooxdoo JSON-RPC were setting protocol 2.0  (after painfully discovering that the client was not, by default, 2.0 compliant).  All this is doing is changing the default so that people who expect JSON-RPC to do what the standard says (2.0) get what they expect.

**Comment by cboulanger** [comment]
Since the tests still all pass, there shouldn't be any backwards-compatibility issues if you rely on the custom extensions to the protocol. The main thing that changes is that all message objects have a "jsonrpc" property with value "2.0" which is required by standard-compliant servers, and which is probabply simply ignored by legacy ones.

**Comment by cboulanger** [comment]
We should, however, insert deprecation warnings for anything non-standard in qooxdoo v6.1 or so, so that people have time to adapt their code, and remove all non-2.0 features in v7.0. The  client could then be moved to a qxl package if anyone needs it to support legacy apps.

**Pull request merged**